### DISCUSSION TITLE
CAN FD implementation

### DISCUSSION THREAD

**Discussion by Chsuma09** [CoRE-RG/FiCo4OMNeT#12] [Ideas]
Hi Team,

Any plans for implementing CANFD?

Any documentation available to implement?

Rgds
Chandra

**Comment by Musteblume** [comment]
Hi Chandra,

we have no plans currently.

Documentation can be generated: Project -> Generate NED Documentation (Uncheck "Source listings (NED, MSG)")
After that there is an index.html in FiCo4OMNeT/doc which you can open to browse the documentation.

Best regards
Philipp

**Reply by Chsuma09** [reply]
Hi Philipp,

Thanks for replying. I have another question regarding "Is there any FiCo4Omnet code bases available with support for bus collisions?"

Rgds
Chandra

**Reply by Musteblume** [reply]
Hi Chandra,

CAN Arbitration is implemented but further than that we have no implementations for collisions that could occur when a node is misbehaving. Is this the type of collisions you are referring to?

Best regards
Philipp
